server/controllers/v1/comment.ts

```diff
@@ -0,0 +1,218 @@
+import { Comment } from '../../models/comment';
+import db from '../../models';
+
+// The return type of a Comment associated with the Post's User.
+export type CommentsUser = Comment & {
+  User: {
+    id: string;
+    firstName: string;
+    lastName: string;
+  };
+};
+
+// The maximum number of results to return.
+const MAX_RESULTS = 10;
+
+export default class CommentController {
+  protected commentsRepo: typeof Comment;
+
+  constructor(commentsRepo: typeof Comment) {
+    this.commentsRepo = commentsRepo;
+  }
+
+  /**
+   * CRUD method to get all the comments on a particular post.
+   *
+   * @param postID  - The identifier used to find the specific post.
+   * @param limit   - Limit the comments to a number of results.
+   * @param offset  - Offset the results
+   * @returns A data object containing the results along.
+   */
+  async getComments(
+    postID: string,
+    limit: number,
+    offset: number
+  ): Promise<{
+    status: number;
+    data: { result?: CommentsUser[]; total?: number; message?: string };
+  }> {
+    const data = await Promise.all([
+      // [0] contains the comments and the user it belongs to
+      this.commentsRepo.findAll({
+        limit: limit > MAX_RESULTS ? MAX_RESULTS : limit,
+        // Since we are returning multiple results, we want to limit the data.
+        attributes: ['id', 'body', 'UserId'],
+        include: [
+          {
+            model: db.User,
+            attributes: ['firstName', 'lastName', 'id'],
+          },
+        ],
+        order: [['createdAt', 'DESC']],
+        offset: offset,
+        where: { PostId: postID },
+      }),
+      // [1] returning the number of comments under this post
+      this.commentsRepo.count({ where: { PostId: postID } }),
+    ]);
+
+    if (!data[0]) {
+      return {
+        status: 404,
+        data: { message: `Post ${postID} could not be found` },
+      };
+    }
+    if (!data[1]) {
```

Comment:
Why not just combine this if statement with the same one on line 59 with an `or`? We are returning `404` for both, so just update the status message with a general error message